### PR TITLE
AC Automated Test - LRAC-8095 | master

### PR DESCRIPTION
**Pull request by 91nilo**
https://issues.liferay.com/browse/LRAC-8095
https://issues.liferay.com/browse/LRAC-8737 (Refactor Ticket)

### CODE REVIEW

**Comment by 91nilo** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/paths/analyticscloud/ACAssets.path [hunk 1]
```diff
@@ -21,6 +21,18 @@
 	<td>//span[@class='title'and text()='${key_metricTitle}']/..//div[@class='metric-value']</td>
 	<td></td>
 </tr>
+
+<!-- CUSTOM ASSETS PAGE -->
+<tr>
+	<td>CUSTOM_ASSET_TITLE</td>
+	<td>//div[contains(@class, 'analytics-custom-metrics-card')]//h5[text()='${customReportName}']</td>
+	<td></td>
+</tr>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/analyticscloud/ACAssets.path:30]: Duplicate with ACCards#CARD_TITLE

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACAssets.macro [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+			locator1 = "TextInput#INPUT_ID",
+			value1 = "${customAssetReportName}");
+
+		Select(
+			locator1 = "Select#SELECT_ID",
+			selectFieldId = "metricSelector",
+			value1 = "${customAssetMetric}");
+
+		Click(locator1 = "Button#SAVE");
+	}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACAssets.macro:21]: Conflict with https://github.com/liferay-qa/liferay-portal/pull/619/commits/259fdcea3060694a21f96409743cb2d167eeab45

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACAssets.macro [hunk 3]
```diff
@@ -1,5 +1,33 @@
...
+
+	macro deleteCustomAssetReport {
+		Click(
+			customReportName = "${customAssetReportName}",
+			locator1 = "ACAssets#CUSTOM_ASSET_DELETE_BUTTON");
+
+		Refresh();
+	}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACAssets.macro:29]: Conflict with https://github.com/liferay-qa/liferay-portal/pull/622/commits/b71bdd9e249e1fdf4970f76a808439ba037130d8

File: portal-web/test/functional/com/liferay/portalweb/paths/analyticscloud/ACAssets.path [hunk 4]
```diff
@@ -21,6 +21,18 @@
...
+<tr>
+	<td>CUSTOM_ASSET_DELETE_BUTTON</td>
+	<td>//div[contains(@class, 'analytics-custom-metrics-card')]//h5[text()='${customReportName}']/../../div//button[@aria-label='Delete']</td>
+	<td></td>
+</tr>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/analyticscloud/ACAssets.path:35]: Conflict with https://github.com/liferay-qa/liferay-portal/pull/622/commits/d87bd45c1059c88dd36a8bff3191be1983f0a01b

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AssetsCustomAsset.testcase [hunk 5]
```diff
@@ -58,6 +58,65 @@ definition {
...
+		var propertyName = ACDXPSettings.connectDXPtoAnalyticsCloud(siteName = "Site Name");
+
+		task ("Add user and access the Custom Asset") {
+			JSONUser.addUser(
+				userEmailAddress = "ac@liferay.com",
+				userFirstName = "ac",
+				userLastName = "ac",
+				userScreenName = "ac");
+
+			ACDXPSettings.syncNewContactData();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AssetsCustomAsset.testcase:73]: Unneeded anymore after https://github.com/liferay-qa/liferay-portal/pull/622 merged.

**Review by Tim-Cao — COMMENTED** [review]
Hi @91nilo, there are some conflict with other PRs. Could you update your case based on top of https://github.com/liferay-qa/liferay-portal/pull/622
No new macros and locators needed.

**Comment by 91nilo** [comment]
ci:test:sf

**Comment by 91nilo** [comment]
@Tim-Cao I did the changes, thank you!
@alexcandidos I could not run this test locally because the Custom Asset doesn't shows up.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AssetsCustomAsset.testcase [hunk 6]
```diff
@@ -170,6 +170,61 @@ definition {
 		}
 	}
 
+	@description = "Feature ID: Legacy | Automation ID: LRAC-8095 | Test Summary: Can delete a custom asset report"
+	@priority = "1"
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AssetsCustomAsset.testcase:174]: @priority = "5"

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AssetsCustomAsset.testcase [hunk 7]
```diff
@@ -170,6 +170,61 @@ definition {
...
+	test CanDeleteCustomAssetReport {
+		task ("Get site assigned property name") {
+			var propertyName = ACDXPSettings.getSiteAssignedPropertyName(siteName = "Site Name");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AssetsCustomAsset.testcase:177]: The `getSiteAssignedPropertyName` is deprecated. Use `getAssignedPropertyName`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AssetsCustomAsset.testcase [hunk 8]
```diff
@@ -170,6 +170,61 @@ definition {
...
+			ACNavigation.switchTab(tabName = "Custom");
+
+			ACAssets.addCustomAssetReport(
+				assetName = "Custom Asset Analytics Cloud",
+				metric = "Asset Views (sum)",
+				reportName = "Report 1");
+		}
+
+		task("Assert that the Custom Asset Report was added"){			
+			AssertElementPresent(
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AssetsCustomAsset.testcase:212]: ACCards. viewCards

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AssetsCustomAsset.testcase (hunk 7 above)
Comment (alexcandidos) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AssetsCustomAsset.testcase:177]: hey @Tim-Cao, Thanks for pointing it out! what do you think about removing the deprecated macro from `ACDXPSettings.macro` as well as its use from the test cases file where we are using it? So we'll avoid people using it... Could you please do that in the next PR?

Thanks!

Comment (Tim-Cao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AssetsCustomAsset.testcase:177]: See https://github.com/liferay-qa/liferay-portal/pull/659/commits/dbeb4f687cbf798ba624d7c8a5610a936e257e7a

Comment (alexcandidos) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AssetsCustomAsset.testcase:177]: Yeaah @Tim-Cao , I just saw you did it on https://github.com/liferay-qa/liferay-portal/pull/659. I'm reviewing and forwarding it there now. Thanks!

**Comment by 91nilo** [comment]
Changed priotiry to 5, and test to use getAssignedPropertyName and viewCards macros.
If there is anything else that needs to be changed just let me know. Thanks!

**Comment by 91nilo** [comment]
ci:test:sf

**Comment by alexcandidos** [comment]
hi @91nilo , as we discussed, in some moments the custom assets is not appearing, so please, proceed with the quarantine/refactor strategy for this case. Thanks!

**Comment by 91nilo** [comment]
ci:test:sf

**Comment by alexcandidos** [comment]
Hi @91nilo , Thanks for working on it!

The code looks good to me!

I'm forwarding it.

**Comment by alexcandidos** [comment]
ci:forward

**Pull request closed without merging**